examples/models/object_detection/retina_net/pascal_voc/train.py

```diff
@@ -0,0 +1,464 @@
+    train_ds,
+    validation_data=eval_ds.take(10),
+    epochs=100,
+    callbacks=callbacks,
+)
+
+"""
+## Evaluation
+"""
+
+coco_suite = [
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        max_detections=100,
+        name="MaP Standard",
+    ),
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        iou_thresholds=[0.75],
+        max_detections=100,
+        name="MaP@IoU=0.75",
+    ),
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        iou_thresholds=[0.5],
+        max_detections=100,
+        name="MaP@IoU=0.5",
+    ),
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        area_range=(0, 32**2),
+        max_detections=100,
+        name="MaP Small",
+    ),
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        area_range=(32**2, 96**2),
+        max_detections=100,
+        name="MaP Medium",
+    ),
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        bounding_box_format="xywh",
+        class_ids=range(len(class_ids)),
+        area_range=(96**2, 1e5**2),
+        max_detections=100,
+        name="MaP Large",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=100,
+        name="Recall Standard",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=1,
+        name="Recall MaxDets=1",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=10,
+        name="Recall MaxDets=10",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=100,
+        area_range=(0, 32**2),
+        name="Recall Small",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=100,
+        area_range=(32**2, 96**2),
+        name="Recall Medium",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(len(class_ids)),
+        bounding_box_format="xywh",
+        max_detections=100,
+        area_range=(96**2, 1e5**2),
+        name="Recall Large",
+    ),
+]
+model.compile(
+    classification_loss=keras_cv.losses.FocalLoss(from_logits=True, reduction="none"),
+    box_loss=keras_cv.losses.SmoothL1Loss(l1_cutoff=1.0, reduction="none"),
+    optimizer=optimizer,
+    metrics=coco_suite,
+)
+
+model.load_weights(CHECKPOINT_PATH)
+
+
+def proc_eval_fn(bounding_box_format, target_size):
```

Comment:
Consider refactoring this with `proc_train_fn`? I didn't really use this other than in the eval.py, so I copied a lot of code for convenience

Reply:
I think most of this will go away with KPLs anyways - should we just focus our efforts there?